models/procurement.py

```diff
@@ -0,0 +1,39 @@
+from odoo import models, _
+from odoo.exceptions import ValidationError
+
+
+class ProcurementGroup(models.Model):
+    _inherit = 'procurement.group'
+
+    def get_group(self, group_identifier, create=False):
+        """
+        Gets the group from the supplied group identifier - either the id
+        or the group name
+        :param group_identifier: The id or the name of the group we are
+            getting or creating
+        :param create: Boolean:  When True and group_identifier is a string
+            will create a new group if it does not exist
+        """
+        name = None
+        if isinstance(group_identifier, int):
+            domain = [('id', '=', group_identifier)]
+        elif isinstance(group_identifier, str):
+            domain = [('name', '=', group_identifier)]
+            name = group_identifier
+        else:
+            raise ValidationError(_(
+                'Unable to create domain for group search from'
+                ' identifier of type %s') % type(group_identifier))
+        results = self.search(domain)
+        if not results:
+            if create and name is not None:
```

Comment:
would it be better to have `if create and name`  what if `group_identifier =  '' ` ?

Reply:
I guess changing "if create and name is not None" to "if create and name" would handle the possibility of the group_identifier being ' '.

However, I got this code from udes_core.models.stock_quant_package.StockQuantPackage#get_package so if we are changing this here does it make sense to change it here as well for consistency?  Because the StockQuantPackage get_package() method could create a package with a string of ' '.

Reply:
Ah ok probably best to check with @ws-mps

Reply:
Yes, change it in both places, we don't want packages or groups with name ''.

Reply:
Update from call.  Only fix this here.  Add a TODO in on StockQuantPackage#get_package and add a task to the backlog?

Reply:
Its only a quick change don't think its worth a US